src/conn/conn_cache.c

```diff
@@ -8,6 +8,32 @@
 
 #include "wt_internal.h"
 
+/*
+ * __cache_config_abs_to_pct --
+ *	Convert the absolute value in cache config to percentage.
+ */
+static inline int
+__cache_config_abs_to_pct(WT_SESSION_IMPL *session,
+    double *param, const char *param_name)
+{
+	WT_CONNECTION_IMPL *conn;
+
+	conn = S2C(session);
+
+	/*
+	 * Anything above 100 is an absolute value; convert it to percentage.
+	 * An absolute value should not exceed the cache size.
+	 */
+	if (*param > 100.0) {
+		if (*param > conn->cache_size)
+			WT_RET_MSG(session, EINVAL,
+			    "%s should be lower than cache size!", param_name);
+		*param = (100.0 * *param) / (conn->cache_size + 1);
```

Comment:
will "*param * 100.0" be more readable. Can we reach here before setting the cache size, I don't think that is possible. cache_size shouldn't be zero here. Maybe we can assert to that instead of adding 1 to cache_size. If planning to keep it this way, please add a comment stating why adding 1 to cache size here.

Reply:
WiredTiger has a provision to provide a cache pool that can be shared between multiple connections. In that scenario the the cache is dynamically allocated for each connection and there are chances of having cache_size 0 for a specific connection. For such cases we want to avoid divide by 0. I will add the comment to indicate that.